examples/read_bids_datasets.py

```diff
@@ -1,82 +1,172 @@
+
+###############################################################################
+# We can now retrieve a list of all MEG-related files in the dataset:
+
+print(bids_path.match())
+
+###############################################################################
+# The returned list contains ``BIDSpaths`` of 3 files:
+# ``sub-01_task-somato_channels.tsv``, ``sub-01_task-somato_events.tsv``, and
+# ``sub-01_task-somato_meg.fif``.
+# The first two are so-called sidecar files that contain information on the
+# recording channels and experimental events, and the third one is the actual
+# MEG data file.
+#
+# Prepare reading the data
+# ------------------------
+#
+# There is obviously only one subject (``01``) and one experimental task
+# (``somato```). Let's use this knowledge to create a new ``BIDSPath`` with
+# all the information required to actually read the MEG data. We also need to
+# pass a ``suffix``, which is the last part of the filename just before the
+# extension -- ``'channels'`` and ``'events'`` for the two TSV files in
+# our example, and ``'meg'`` for MEG raw data. For MEG and EEG raw data, the
+# suffix is identical to the datatype, so don't let yourselve be confused here!
+
 bids_root = somato.data_path()
-subject_id = '01'
-task = 'somato'
 datatype = 'meg'
+subject = '01'
+task = 'somato'
+suffix = 'meg'
+
+bids_path = BIDSPath(subject=subject, task=task, suffix=suffix,
+                     datatype=datatype, root=bids_root)
 
-bids_path = BIDSPath(subject=subject_id, task=task,
-                     datatype=datatype, suffix=datatype,
-                     root=bids_root)
+###############################################################################
+# Now let's print the contents of ``bids_path``.
 
-# bids basename is nicely formatted
 print(bids_path)
 
 ###############################################################################
-# Print the directory tree
-print_dir_tree(bids_root)
+# You probably noticed two things: Firstly, this looks like an ordinary string
+# now, not like the more-or-less neatly formatted output we saw before. And
+# secondly, that there's suddenly a filename extension which we never specified
+# anywhere!
+#
+# The reason is that when you call ``print(bids_path)``, ``BIDSPath`` returns
+# a string representation of ``BIDSPath.fpath``, which looks different. If,
+# instead, you simply typed ``bids_path`` (or ``print(repr(bids_path))``, which
+# is the same) into your Python console, you would get the nicely formatted
+# output:
+
+bids_path
 
 ###############################################################################
-# Step 2: Read a BIDS dataset
-# ---------------------------
+# But what about that filename extension, now? ``BIDSPath.fpath``, which –
+# as you hopefully remember – is invoked when you to ``print(bids_path)`` –
+# employs some heuristics to auto-detect some missing filename components.
+# This can come in handy if, for example, you intend to re-use the same
```

Comment:
This feels like a very niche example that should be left out of the first example. I'd rather just "show and tell"

Reply:
I mean the example of using a "different acquisition system"